blockchain/state_manager/src/lib.rs

```diff
@@ -69,18 +96,20 @@ where
     /// constructor
     pub fn new(bs: Arc<DB>) -> Self {
         Self {
-            bs,
+            cs: ChainStore::new(bs),
```

Comment:
You can't just create a ChainStore that is independent from the actual `ChainStore` and expect it to work. If you need to have a chainstore attached here, you should be passing it in as an `Arc` of the actual one, to avoid duplication and bugs with that